qdata/parser.py

```diff
@@ -1,38 +1,47 @@
+"""This module contains the QASMToIRTransformer and qasm_parser"""
 from enum import Enum
+import pathlib
+
 import sympy
 
 from lark import Lark, Transformer
 
-from parser_utils import *
-
-with open("qasm.lark", "r") as f:
+from .ir import (
+    QasmProgram,
+    Lists,
+    Ops,
+    Declarations,
+    ParsedList,
+    unpack,
+    flatten,
+    format_wires,
+    ArithmeticOperation,
+    UnaryOperation,
+    BinaryOperation,
+    Op,
+    Gate,
+    Channel,
+    Operator,
+    TensorOp,
+    Term,
+    Measure,
+    Barrier,
+    EqualityCondition,
+    ConditionalOp,
+    Declaration,
+    ClassicalRegister,
+    QuantumRegister,
+    GateDeclaration,
+    OperatorDeclaration
```

Comment:
It's great having these elsewhere, but there are still a lot. Wonder if we should try to simplify these further? Maybe by having an `ir` submodule?

Reply:
Yep, I agree, definitely needs further simplification.

Note that not all of these are actually required in this file, after blacking I can run pylint to see which ones are 'internal'.